File: cs/sample/Sample.Tests/SumOfLeftLeaves/testSumOfLeftLeaves.cs
using System;
using Xunit;
using Sample.SumOfLeftLeaves;
using Sample.TreeNode;

namespace Sample.Tests{
    public class UnitTest_SumOfLeftLeaves{
        [Theory]
        [InlineData(new string[]{"3","9","20","None","None","15","7"},24)]
        [InlineData(new string[]{"1"},0)]
        public void checkResult(string[] nums,int expected){
            treeNode root = treeNode.treeNode_fromList2(nums);
            Assert.Equal(expected,sumOfLeftLeaves.naive(root));
        }
    }
}

File: py/tests/testSumOfLeftLeaves/test_SumOfLeftLeaves.py
from solns.treeNode.treeNode import *
import unittest
from parameterized import parameterized as p
from solns.sumOfLeftLeaves.sumOfLeftLeaves import *

class UnitTest_SumOfLeftLeaves(unittest.TestCase):
    @p.expand([
        [[3,9,20,None,None,15,7],24],[[1],0]
    ])
    def test_naive(self,nums,expected):
        root = TreeNode.treeNode_fromList2(nums)
        self.assertEqual(Solution.naive(root),expected)
